button-merchant/src/main/java/com/usebutton/merchant/ConnectionManagerImpl.java

```diff
@@ -107,13 +109,16 @@ public NetworkResponse executeRequest(@NonNull ApiRequest request)
             if (responseCode >= 400) {
                 String message = "Unsuccessful Request. HTTP StatusCode: " + responseCode;
                 Log.e(TAG, message);
-                throw new ButtonNetworkException(message);
+                throw new HttpStatusException(message, responseCode);
             }
 
             JSONObject responseJson = readResponseBody(urlConnection);
             return new NetworkResponse(responseCode, responseJson);
-        } catch (IOException | JSONException e) {
-            Log.e(TAG, e.getClass().getSimpleName() + " has occurred", e);
+        } catch (IOException e) {
```

Comment:
I would recommend having `getConnection()` throw the `ButtonNetworkException`. That way we can be certain that the source of that one `IOException` does indeed pertain to network not being found. No need to catch it here then.